Assets/MapGenerator.cs

```diff
@@ -0,0 +1,274 @@
+			{
+				map[x,y] = newMap[x,y];
+			}
+		}
+	}
+
+	void SeedCaves()	//generate a number of 'holes' in the ground where caves could form later
+	{
+		if (useRandomSeed) 
+		{
+			seed = Time.time.ToString();
+		}
+		System.Random pseudoRandom = new System.Random(seed.GetHashCode());
+		for (int x = 1; x < width-1; x ++)  //Don't seed the edges
+		{
+			//for (int y = 0; y < map[x].Topography-1; y ++) //stops 1 block short of ground surface
+			for (int y = 3; y < Topography[x]-5; y ++) //stops 5 block short of ground surface
+			{
+				//if(map[x].Column[y] != 0)
+				if(map[x,y] != 0)
+				{
+					//should we seed a cave here?
+					if(pseudoRandom.Next(0,100) < randomCaveSeedPercent)
+					{
+						//map[x].Column[y] = 0;
+						map[x,y] = 0;
+					}
+				}
+			}
+		}
+	}
+
+
+	void SmoothMap() 
+	{
+		newMap = new int[width,height];
+		for (int x = 0; x < width; x ++) 
+		{
+			for (int y = 0; y < height; y ++) 
+			{
+				int neighbourWallTiles = GetSurroundingWallCount(x,y);
+				if(x > 0 && x  < width -1 && y > 0 && y < Topography[x] - 5)  //do not process any cells that are out of bounds
+				{
+					if(map[x,y] != 0) //Location IS a wall. Uses != 0 rather than == 1 to allow for multiple values for walls in the future
+					{
+						if (neighbourWallTiles < DeathLimit)
+						{
+							//map[x].Column[y] = 0;
+							newMap[x,y] = 0;
+						}
+						else
+						{
+							newMap[x,y] = 1;
+						}
+					}
+					else  //Location IS NOT a wall
+					{
+						if (neighbourWallTiles > BirthLimit)  //Does it have enough neighbors to become a wall?
+						{
+							//map[x].Column[y] = 1;
+							newMap[x,y] = 1;
+						}
+						else
+						{
+							newMap[x,y] = 0;
+						}
+					}
+				}
+				else //if out of bounds then just copy in values from map instead
+				{
+					newMap[x,y] = map[x,y];
+				}
+			}
+		}
+		//map = newMap;
+		SwapMaps ();
+	}
+
+	int GetSurroundingWallCount(int gridX, int gridY) 
+	{
+		int wallCount = 0; //maximum possible  = 8
+		for (int neighbourX = gridX - 1; neighbourX <= gridX + 1; neighbourX ++) 
+		{
+			for (int neighbourY = gridY - 1; neighbourY <= gridY + 1; neighbourY ++) 
+			{
+				if (neighbourX >= 0 && neighbourX < width && neighbourY >= 0 && neighbourY < height) //excludes edge tiles to prevent crashes
+				{
+					if (neighbourX != gridX || neighbourY != gridY) //do not count our own centre tile at X,Y
+					{
+						//wallCount += map[neighbourX].Column[neighbourY];
+						wallCount += map[neighbourX, neighbourY];
+					}
+				}
+				else 
+				{
+					wallCount ++;
+				}
+			}
+		}
+
+		return wallCount;
+	}
+
+
+	string GetCellValue(int gridX, int gridY)
+	{
+		string CellValue = "0";
+		if (gridX > 1 && gridX < width - 2) 
+		{
+			if (gridY > 1 && gridY < Topography[gridX] - 5) //this logic might seem weird but we cannot check Topography unless we know X is within bounds first
+			{
+				{
+					if (map [gridX, gridY] != 0) //it's a wall of some kind
+					{ 
+						CellValue = "1";
```

Comment:
The 1 in this case has to be a single byte in the form of a string so a longer more descriptive name is not possible. The "1" and "0" values from this function are concatenated into an 8 byte string so that binary logic can be performed on it.
However, it's a moot point as this entire function needs to be deleted This method failed to produce results anyway